sources/metricdog/src/metricdog.rs

```diff
@@ -0,0 +1,158 @@
+    config: Config,
+    /// Information about the Bottlerocket release, e.g. from `os-release`
+    os_release: BottlerocketRelease,
+    /// A trait object that checks if a service (listed in `config`) is healthy. This can be passed-
+    /// in, but defaults to an object that uses `systemctl` to check services.
+    healthcheck: Box<dyn ServiceCheck>,
+    /// The metrics_url, having been parsed during construction of the `Metricdog` object.
+    metrics_url: Url,
+}
+
+impl Metricdog {
+    /// Create a new instance by optionally passing in the `Config`, `BottlerocketRelease`, and
+    /// `ServiceCheck` objects. For each of these, if `None` is passed, then the default is used.
+    pub(crate) fn from_parts(
+        config: Config,
+        os_release: BottlerocketRelease,
+        healthcheck: Box<dyn ServiceCheck>,
+    ) -> Result<Self> {
+        let metrics_url = Url::from_str(&config.metrics_url).context(error::UrlParse {
+            url: &config.metrics_url,
+        })?;
+        Ok(Self {
+            config,
+            os_release,
+            healthcheck,
+            metrics_url,
+        })
+    }
+
+    /// # Description
+    ///
+    /// Sends key-value pairs as query parameters in a GET request to the URL in `config`. A
+    /// standard set of key-value pairs are added first, and appended by any additional parameters
+    /// passed in to this function.
+    ///
+    /// # Parameters
+    ///
+    /// * `sender`:          This is the name of the application sending the metrics e.g.
+    ///                      `metricdog` or `updog`.
+    /// * `event`:           The name of the type of metrics event that is being sent. For example
+    ///                      `boot_success` or `health_ping`.
+    /// * `values`:          The key-value pairs that you want to send. These will be sorted by key
+    ///                      before sending to ensure consistency of key-value ordering.
+    /// * `timeout_seconds`: The timeout setting for the HTTP client. Defaults to
+    ///                      `DEFAULT_TIMEOUT_SECONDS` when `None` is passed.
+    pub(crate) fn send<S1, S2>(
+        &self,
+        sender: S1,
+        event: S2,
+        values: Option<&HashMap<String, String>>,
+        timeout_seconds: Option<u64>,
+    ) -> Result<()>
+    where
+        S1: AsRef<str>,
+        S2: AsRef<str>,
+    {
+        let mut url = self.metrics_url.clone();
+        {
+            let mut q = url.query_pairs_mut();
+            q.append_pair("sender", sender.as_ref());
+            q.append_pair("event", event.as_ref());
+            q.append_pair("version", &self.os_release.version_id.to_string());
+            q.append_pair("variant", &self.os_release.variant_id);
+            q.append_pair("arch", &self.os_release.arch);
+            q.append_pair("aws_region", &self.config.aws_region);
+            q.append_pair("seed", format!("{}", &self.config.seed).as_str());
```

Comment:
```suggestion
            q.append_pair("seed", self.config.seed.to_string().as_str());
```
And maybe we don't need `as_str()`, given that `ignore_waves` below is passing a `String` directly?

Reply:
Fixed in [b8b5c3a](https://github.com/bottlerocket-os/bottlerocket/compare/153acff977082318fef8fc8e61024b26574c362e..b8b5c3ad92298db2c46262eb2b6c5b6ea3f60187)